anchor-service/src/anchor_batch.rs

```diff
@@ -0,0 +1,99 @@
+use std::{sync::Arc, time::Duration};
+
+use anyhow::Result;
+use async_trait::async_trait;
+use cid::Cid;
+use indexmap::IndexMap;
+use tokio::time::interval;
+use tracing::{error, info};
+
+use ceramic_event::anchor::{AnchorRequest, TimeEventBatch};
+
+use crate::{
+    merkle_tree::{build_merkle_tree, MerkleTree},
+    time_event::build_time_events,
+    transaction_manager::{Receipt, TransactionManager},
+};
+
+#[async_trait]
+pub trait AnchorClient: Send + Sync {
```

Comment:
Who is expected to implement this trait? Something from the ceramic-service crate?

If so I think we should slightly tweak the semantics of the trait. Instead of AnchorClient we should call this FeedClient and instead of `get_anchor_requests` we should have `get_events(FeedOpts)` where FeedOpts lets you decide which kind of events to get from the Feed. This way the `ceramic-service` does not have to know or understand anything about anchors. Rather it remains specific to Ceramic events.  

This likely also means that the AnchorRequest type can remain private to the anchor-service crate.


Thoughts?

Reply:
This feels worth a design discussion.